Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/jxbrowserutils/BrowserUtils.java

```diff
@@ -0,0 +1,273 @@
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.math.BigInteger;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.security.MessageDigest;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Executors;
+import java.util.concurrent.FutureTask;
+
+import static java.lang.Thread.sleep;
+
+/**
+ * Created by t-rufang on 7/14/2017.
+ */
+class MyException extends Exception {
+    public MyException() {
+        super();
+    }
+
+    public MyException(String message) {
+        super(message);
+    }
+}
+
+class HDInsightHelperTest implements HDInsightHelper {
+
+    public void openJobViewEditor(@NotNull Object projectObject, @NotNull String uuid) {}
+    public void closeJobViewEditor(@NotNull Object projectObject, @NotNull String uuid) {}
+    public String getPluginRootPath() {
+        return ".";
+    }
+    public String getInstallationId() {
+        return "";
+    }
+    public boolean isIntelliJPlugin() {
+        return true;
+    }
+}
+
+
+public class BrowserUtils {
+    private static final ImmutableMap<String, String> map = ImmutableMap.of(
+            "win32", "jxbrowser-win32-6.16",
+            "win64", "jxbrowser-6.16",
+            "mac", "jxbrowser-mac-6.16",
+            "linux64", "jxbrowser-linux64-6.16",
+            "linux32", "jxbrowser-linux64-6.16");   // TODO: jxbrowser does't provide linux32 jar, we provide linux64 instead
+    public static final int MAX_DOWNLOAD_FAILURE_RETRY_TIMES = 3;
+    public static final int MAX_CHECK_FAILURE_RETRY_TIMES = 3;
+    public static final String storageConnectionString =
+            "DefaultEndpointsProtocol=https;"
+                    + "AccountName=ltiantest;"
+                    + "AccountKey=UBCWXXk1ha9ktwa5WTFN0MNSQOGMnyFg5KQvruM+AX4nQHxIQ7LjpTTIQtAM7ZsRbkhayUqy1AT80V+ZPssQFA==";
+    private static final Logger logger = LoggerFactory.getLogger(BrowserUtils.class);
+
+    public static boolean md5Check(String verifyFileName, String md5FileName) throws Exception {
+        File verifyFile = new File(verifyFileName);
+        File md5File = new File(md5FileName);
+        if (!verifyFile.exists()) {
+            throw new MyException(verifyFileName + " does not exist.");
+        }
+        if (!md5File.exists()) {
+            throw new MyException(md5FileName + " does not exist.");
+        }
+
+        FileInputStream in = null;
+        Scanner scanner = null;
+        try {
+            // calculate md5 from jar file
+            MessageDigest md5 = MessageDigest.getInstance("MD5");
+            in = new FileInputStream(verifyFileName);
+            byte[] buffer = new byte[4096];
+            int length = -1;
+            while ((length = in.read(buffer)) != -1) {
+                md5.update(buffer, 0, length);
+            }
```

Comment:
org.apache.commons.codec.digest.DigestUtils is suggested here. You can have try.